dandi/upload.py

```diff
@@ -413,8 +413,14 @@ def _bids_discover_and_validate(
     from .utils import find_files
     from .validate import validate_bids
 
+    lgr.debug("Discovering root directories of BIDS datasets")
     bids_descriptions = map(
-        Path, find_files(r"(^|[/\x5C])dataset_description\.json$", dandiset_path)
+        Path,
+        find_files(
+            r"(^|[/\x5C])dataset_description\.json$",
+            dandiset_path,
+            dirs_avoid=r"\.(ngff|zarr)$",
```

Comment:
ha -- the case at hands has `micr/sub-MITU01h3_ses-20211202h20m54s11_sample-20_stain-NN_run-1_chunk-4_SPIM.ngff.source` so I would need to add allowing for more extensions for such cases.

Reply:
Let's allow for any other suffix as well:

```suggestion
            dirs_avoid=r"\.(ngff|zarr)(\..*)?$",
```

Reply:
Note that the main upload code currently only regards directories ending in `.ngff` and `.zarr` as Zarrs; `*.ngff.source` and the like are just treated as regular directories.

Reply:
oh.  Thanks! prepared separate PR for that: https://github.com/dandi/dandi-cli/pull/1077 .